### PR TITLE
On chain wiki

### PR DESCRIPTION
**Pull request by Akado2009**

### CODE REVIEW

File: docs/examples/onchain-nft.md [hunk 1]
```diff
@@ -0,0 +1,361 @@
+---
+sidebar_label: OnChain NFT
+sidebar_position: 14
```
Comment (AndrePanin) [inline docs/examples/onchain-nft.md:3]:
```suggestion
sidebar_position: 16
```

File: docs/examples/onchain-nft.md [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+---
+
+# Gear Non-Fungible Token
```
Comment (AndrePanin) [inline docs/examples/onchain-nft.md:6]:
```suggestion
# On-chain gNFT assets
```

File: docs/examples/onchain-nft.md [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+### Introduction
```
Comment (AndrePanin) [inline docs/examples/onchain-nft.md:8]: Pls keep only essential info about gNFT, for the rest - provide a link on existing page: https://wiki.gear-tech.io/examples/gnft-721

File: docs/examples/onchain-nft.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+
+When the owner of a given token ID wishes to transfer it to another user, it is easy to verify ownership and reassign the token to a new owner.
+
+Mostly NFTs images (or other base resourse) is stored somewhere else (e.g. ipfs) and only the metadata is stored in the contract. Metadata consists of a name, an ID and links to the external resources, where the images is actually stored.
+But there is another approached introduced here. Sometimes you can store NFTs directly on chain without any external storage. This approach helps you not to lose your nft if there is a problem with the external storage.
+
+### Approach
+To successfully implement this approach several things are needed. Firstly, when initializing a collection, one should provide all the possible images of all the layers for a collection. Secondly, when minting alongside with a small metadata, one should provide a combination of layers used for a specific NFT. This approach seems quite costly when initializing, but is relatively cheap when it comes to minting.
+
+### Default non-fungible-token implementation
```
Comment (AndrePanin) [inline docs/examples/onchain-nft.md:20]: Show the specific of on-chain NFT, keep only necessary info that needs to understand the difference of common NFT implementation from on-chain NFT.

**Review by AndrePanin — CHANGES_REQUESTED** [review]

**Comment by shamilsan** [comment]
Stale PR? @Akado2009

File: docs/examples/onchain-nft.md (hunk 3 above)
Comment (Akado2009) [reply docs/examples/onchain-nft.md:8]: Fixed.

**Comment by Akado2009** [comment]
> Stale PR? @Akado2009
Fixed, needs a review.

File: docs/examples/onchain-nft.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+# On-chain gNFT assets
+
+### Introduction
+On the info related to gNFT consider reading: [gNFT-721](https://wiki.gear-tech.io/examples/gnft-721).
+
+When the owner of a given token ID wishes to transfer it to another user, it is easy to verify ownership and reassign the token to a new owner.
+
+Mostly NFTs images (or other base resourse) is stored somewhere else (e.g. ipfs) and only the metadata is stored in the contract. Metadata consists of a name, an ID and links to the external resources, where the images is actually stored.
+But there is another approached introduced here. Sometimes you can store NFTs directly on chain without any external storage. This approach helps you not to lose your nft if there is a problem with the external storage.
```
Comment (shamilsan) [inline docs/examples/onchain-nft.md:14]:
```suggestion
But there is another approach introduced here. Sometimes you can store NFTs directly on chain without any external storage. This approach helps you not to lose your ТАЕ if there is a problem with the external storage.
```

File: docs/examples/onchain-nft.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+---
+
+# On-chain gNFT assets
+
+### Introduction
+On the info related to gNFT consider reading: [gNFT-721](https://wiki.gear-tech.io/examples/gnft-721).
+
+When the owner of a given token ID wishes to transfer it to another user, it is easy to verify ownership and reassign the token to a new owner.
+
+Mostly NFTs images (or other base resourse) is stored somewhere else (e.g. ipfs) and only the metadata is stored in the contract. Metadata consists of a name, an ID and links to the external resources, where the images is actually stored.
```
Comment (shamilsan) [inline docs/examples/onchain-nft.md:13]:
```suggestion
Mostly NFTs images (or other base resources) are stored somewhere else (e.g. IPFS) and only the metadata is stored in the contract. Metadata consists of a name, an ID and links to the external resources, where the images are actually stored.
```

File: docs/examples/onchain-nft.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+        }
+    }
+}
+```
+
+## Conclusion
+
+Gear provides a reusable [library](https://github.com/gear-dapps/non-fungible-token/tree/master/nft/src) with core functionality for the gNFT protocol. By using object composition, that library can be utilized within a custom NFT contract implementation in order to minimize duplication of community available code.
+
+A source code of the NFT marketplace provided by Gear is available on GitHub: [on-chain-nft/src](https://github.com/gear-dapps/non-fungible-token/tree/master/on-chain-nft/src).
```
Comment (shamilsan) [inline docs/examples/onchain-nft.md:251]: marketplace?

**Review by shamilsan — APPROVED** [review]

**Review by AndrePanin — APPROVED** [review]

**Pull request merged**